src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.StateMachineMethodToClassRewriter.vb

```diff
@@ -141,33 +155,77 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End Sub
             End Structure
 
-            Protected Function AddState() As StateInfo
-                Dim stateNumber As Integer = Me.NextState
-                Me.NextState += 1
+            Protected Sub AddResumableState(awaitOrYieldReturnSyntax As SyntaxNode, <Out> ByRef stateNumber As Integer, <Out> ByRef resumeLabel As GeneratedLabelSymbol)
+                AddResumableState(_resumableStateAllocator, awaitOrYieldReturnSyntax, stateNumber, resumeLabel)
+            End Sub
+
+            Protected Sub AddResumableState(allocator As ResumableStateMachineStateAllocator, awaitOrYieldReturnSyntax As SyntaxNode, <Out> ByRef stateNumber As Integer, <Out> ByRef resumeLabel As GeneratedLabelSymbol)
```

Comment:
I think the only caller for this overload is above at line 159. If that's correct, consider inlining (use `_resumableStateAllocator` directly, instead of introducing a parameter `allocator` for it).

Reply:
Gentle ping. What did you think of this suggestion?

Reply:
Sorry, missed this one. Yes, in VB it only has a single caller since VB does not have async iterators. I matched the VB implementation to C#, but I can inline.